routing/router.go

```diff
@@ -3363,3 +3363,49 @@ func forwardPass(senderAmt lnwire.MilliSatoshi,
 
 	return senderAmt, nil
 }
+
+// outgoingFromIncoming computes the outgoing amount based on the incoming
+// amount by subtracting fees from the incoming amount. Note that this is not
+// exactly the inverse of incomingFromOutgoing, because of the rounding.
+func outgoingFromIncoming(incomingAmt lnwire.MilliSatoshi,
+	incoming, outgoing *unifiedEdge) lnwire.MilliSatoshi {
+
+	// TODO: calculations with integer ceil/floor arithmetic
+	feeRateParts := 1e6
```

Comment:
I think we can just keep everything multiplied by a million and then do integer division later on?
Not sure about the requirements of the actual algorithm, I just know we should try to avoid `float64` wherever possible!
But I guess that's exactly what the TODOs here are for.

Reply:
It was kind of tough to do integer arithmetic here because formulas involve products of PPM*PPM*amount, which can easily overflow a int64 and we need to handle negative numbers as well. I tried an approach with floats, why at least don't have a problem overflowing when adding/multiplying numbers and I think there are no problems related to accumulating errors, but not too knowledgeable here. Another option would be to use `big.Int`, but the syntax there looks very clumsy